### PR TITLE
[bsp]: add qemu versatilepb and fix x86 build errors

### PR DESCRIPTION
**Pull request by parai**
1.add qemu machine versatilepb, plan to be used as a base to add lwext4 filesystem
2.fix x86 build errors as 3.0 update

### CODE REVIEW

**Comment by parai** [comment]
Currently, there is issue on versatilepb msh, when input command. a exception as below log shows, but if create 2 period task with message print out, the msh can be used, very strange, hint me what is the issue for fast solve it out.

 \ | /
- RT -     Thread Operating System
 / | \     3.0.0 build Nov  7 2017
 2006 - 2017 Copyright by rt-thread team
msh >
Execption:
r00:0x00025b4c r01:0xffffffff r02:0x01030887 r03:0x00000001
r04:0x00025b4c r05:0xdeadbeef r06:0xdeadbeef r07:0xdeadbeef
r08:0xdeadbeef r09:0xdeadbeef r10:0xdeadbeef
fp :0x0103087c ip :0x00025abc
sp :0x00026380 lr :0x00028004 pc :0x00028004
cpsr:0x20000013
undefined instruction
thread - tshell stack:
thread pri  status      sp     stack size max used left tick  error
------ ---  ------- ---------- ----------  ------  ---------- ---
tidle   31  ready   0x00000070 0x00000400    13%   0x00000011 000
tshell  20  ready   0x000000d8 0x00000800    10%   0x0000000a 000
shutdown...

File: bsp/x86/applications/SConscript [hunk 1]
```diff
@@ -4,7 +4,8 @@ from building import *
 
 cwd     = os.path.join(str(Dir('#')), 'applications')
 src	= Glob('*.c')
-CPPPATH = [cwd, str(Dir('#'))]
+drvinc=os.path.abspath('%s/../../../components/drivers/include'%(cwd))
```
Comment (BernardXiong) [inline bsp/x86/applications/SConscript]: please not use this way to handle path.

File: bsp/x86/applications/application.c [hunk 2]
```diff
@@ -84,3 +84,8 @@ int rt_application_init()
 
     return 0;
 }
+
```
Comment (BernardXiong) [inline bsp/x86/applications/application.c]: Which libc is used for x86?

File: libcpu/ia32/__udivsi3.c [hunk 3]
```diff
@@ -14,12 +14,9 @@
 
 #include <rtthread.h>
 
-typedef rt_uint32_t uint32_t;
```
Comment (BernardXiong) [inline libcpu/ia32/__udivsi3.c]: Or to include <stdint.h>?

**Comment by BernardXiong** [comment]
Please split your PR.

File: bsp/x86/applications/SConscript (hunk 1 above)
Comment (parai) [reply bsp/x86/applications/SConscript]: but need to include rtdevice.h by DFS, how to

File: bsp/x86/applications/application.c (hunk 2 above)
Comment (parai) [reply bsp/x86/applications/application.c]: seems by default minilibc

File: libcpu/ia32/__udivsi3.c (hunk 3 above)
Comment (parai) [reply libcpu/ia32/__udivsi3.c]: good idea

File: bsp/x86/applications/SConscript (hunk 1 above)
Comment (BernardXiong) [reply bsp/x86/applications/SConscript]: #define RT_USING_DEVICE_IPC
in rtconfig.h

**Pull request closed without merging**